app/services/payment/spam_protection_service.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module Payment
+  class SpamProtectionService
+    def initialize(account, payment_status, user_session = nil)
+      @account         = account
+      @payment_status  = payment_status
+      @gateway_setting = account.gateway_setting
+      @user_session    = user_session
+    end
+
+    def call
```

Comment:
imo this should return a truthy value if it works and a falsy one if it doesn't. But it isn't a big deal in this case because it is only used in a callback without caring about the response

Reply:
I didn't add any return because it wasn't needed in the usage of it.

Reply:
Right, I agree there is no need for now but I mentioned it because it does not add any complexity and can be handy.
Like:
```ruby
def call
  return true if payment_successful

  limiter = ActionLimiter.new(@account)
  limiter.perform 'invalid_payment'
rescue ActionLimiter::ActionLimitsExceededError
  account.force_suspend!
  user_session&.revoke!
  Rails.logger.info "spam_protection_service: account_suspended id: #{account.id}"
  nil
end
```

But as you said, there is no need so feel free to don't do it :wink:

Reply:
```ruby
return true if payment_successful
```

is confusing IMO. I'd expect the service to return true if it did something.

Reply:
@guicassolato that makes sense too. then:

```ruby
def call
  return if payment_successful

  limiter = ActionLimiter.new(@account)
  limiter.perform 'invalid_payment'
rescue ActionLimiter::ActionLimitsExceededError
  account.force_suspend!
  user_session&.revoke!
  Rails.logger.info "spam_protection_service: account_suspended id: #{account.id}"
  nil
end
```

As: `nil` means that it didn't do anything, truthy means that it did (for the limiter.perform)